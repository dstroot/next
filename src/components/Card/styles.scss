.nxt_card-width {
  width: 90%;
  max-width: 24rem;
}

.nxt_rounded-corner {
  border-radius: 12px;
}
