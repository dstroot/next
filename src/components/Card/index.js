import React from 'react';
import './styles.scss';

class Card extends React.Component {
  render() {
    return (
      <div className="card">
        <div className="card-body">
          <img src={this.props.img} alt="Evergreen" className="mb-2" />
          <h5 className="card-title nxt_heading-small mb-4">
            {this.props.title}
          </h5>
          <p className="card-text circular mb-5">{this.props.content}</p>
          <div>{this.props.button}</div>
        </div>
      </div>
    );
  }
}

export default Card;
