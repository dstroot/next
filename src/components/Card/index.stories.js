import React from 'react';
import { storiesOf } from '@storybook/react';
import Card from '../Card';

storiesOf('NEXT|sections|Card', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show Card', () => <Card />);
