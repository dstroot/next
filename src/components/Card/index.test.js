import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import Card from '../Card';

describe('Card', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Card />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<Card />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
