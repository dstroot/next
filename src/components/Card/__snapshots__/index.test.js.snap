// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card it should render 1`] = `
<div
  className="card h-100"
>
  <div
    className="card-body py-4"
  >
    <img
      alt="Evergreen"
      className="mb-2"
    />
    <div
      className="card-title nxt_heading-small mb-4"
    />
    <div
      className="card-text circular mb-5"
    />
    <div />
  </div>
</div>
`;
